org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleCryptosystemTextbook.java

```diff
@@ -0,0 +1,974 @@
+			return;
+		}
+		
+		hideControl(cstb.txtDecryptWarning);
+		
+		ciphertextList = guiHandler.rabinFirst.parseString(ciphertext, guiHandler.blocklength);
+		try {
+			plaintexts = guiHandler.rabinFirst.getAllPlaintextsFromListOfCiphertextblocks(ciphertextList, guiHandler.radix);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			cstb.txtEncryptionWarning.setText("Attention: the computation takes too much time. Please generate a greater N.");
+			guiHandler.showControl(cstb.txtEncryptionWarning);
+
+		}
+		initializeClickedPlaintexts(plaintexts.size(), possiblePlaintextsSize);
+		String ciphertextblocksWithSeparator = guiHandler.rabinFirst.getStringWithSepForm(ciphertextList, guiHandler.separator);
+		cstb.txtCiphertextSegments.setText(ciphertextblocksWithSeparator);
+		
+		setChooseBlock(cstb.cmbChooseBlock, plaintexts.size());
+		cstb.cmbChooseBlock.select(0);
+		
+		
+		int idx = cstb.cmbChooseBlock.getSelectionIndex();
+		int elem = idx + 1;
+		markCiphertext(cstb.txtCiphertextSegments, elem);
+			
+		setPossiblePlaintexts(idx, cstb.txtFirstPlaintext, cstb.txtSecondPlaintext, 
+				cstb.txtThirdPlaintext, cstb.txtFourthPlaintext);
+		
+		updateChosenPlaintexts(idx, cstb, GUIHandler.colorSelectControlBG, GUIHandler.colorDeselectControlBG, GUIHandler.colorSelectControlFG, GUIHandler.colorDeselectControlFG);
+		currentSelectedPlaintexts = new LinkedHashMap<String, String>();
+		cstb.txtChosenPlaintexts.setText("");
+	}
+		
+	
+	
+	private void resetChosenPlaintexts(int idx, CryptosystemTextbookComposite cstb) {
+		updateChosenPlaintexts(idx, cstb, GUIHandler.colorSelectControlBG, GUIHandler.colorDeselectControlBG, GUIHandler.colorSelectControlFG, GUIHandler.colorDeselectControlFG);
+		cstb.txtChosenPlaintexts.setText("");
+	}
+	
+	
+	
+	private String getCurrentSelectedPlaintexts() {
+		String ret = "";
+		Set<String> keys = currentSelectedPlaintexts.keySet();
+		for(String key : keys) {
+			ret += currentSelectedPlaintexts.get(key);
+		}
+		return ret;
+	}
+	
+	
+	
+	private void setColorPlaintext(CryptosystemTextbookComposite cstb, int idxToSet, Color colorBG, Color colorFG) {
+		switch(idxToSet) {
+			case 0:
+				cstb.txtFirstPlaintext.setBackground(colorBG);
+				cstb.txtFirstPlaintext.setForeground(colorFG);
+				break;
+			
+			case 1:
+				cstb.txtSecondPlaintext.setBackground(colorBG);
+				cstb.txtSecondPlaintext.setForeground(colorFG);
+				break;
+			
+			case 2:
+				cstb.txtThirdPlaintext.setBackground(colorBG);
+				cstb.txtThirdPlaintext.setForeground(colorFG);
+				break;
+				
+			case 3:
+				cstb.txtFourthPlaintext.setBackground(colorBG);
+				cstb.txtFourthPlaintext.setForeground(colorFG);
+				break;
+			
+			default:
+				break;
+		}
+	}
+	
+	
+	
+	private boolean isOnePlaintextSelected(int idx) {
+		ArrayList<Boolean> clickedPlaintextsAtIdx = clickedPlaintexts.get(idx);
+		for(boolean b : clickedPlaintextsAtIdx) {
+			if(b == true)
+				return true;
+		}
+		
+		return false;
```

Comment:
Streams can save a lot of verbosity in such a case as well:
```suggestion
	return clickedPlaintexts.get(idx).stream().anyMatch((x)-> x == true);
```